deployments_plan_for_CEX/readme.md

```diff
@@ -0,0 +1,251 @@
+- restart back nodeos, eos-evm-node & eos-evm-rpc
+- for leap recovery, please restore the data-dir folder of the last backup and use the leap's snapshot
+- for eos-evm-node recovery, please restore the chain-data folder of the last backup.
+
+<a name="RMS"></a>
+## Running the eos-evm-miner service 
+The miner service will help to package the EVM transaction into EOS transaction and set to the EOS network. It will provide the following 2 eth API:
+- eth_gasPrice: retrieve the currect gas price from EOS Network
+- eth_sendRawTransaction: package the ETH transaction into EOS transaction and push into the EOS Network.
+clone the https://github.com/eosnetworkfoundation/eos-evm-miner repo
+
+- create your miner account (for example: a123) on EOS Network
+- open account balance on EVM side:
+  ```
+  ./cleos push action eosio.evm open '{"owner":"a123"}' -p a123
+  ```
+- powerup the miner account with enough CPU & NET resource (for example: 1min CPU. 10 MB net per day). You can use some existing auto powerup service such as https://eospowerup.io/auto or push the powerup transaction (eosio::powerup) via cleos.
+
+- prepare the .env file with the correct information
+```
+PRIVATE_KEY=5KQwrPbwdL6PhXujxW37FSSQZ1JiwsST4cqQzDeyXtP79zkvFD3
+MINER_ACCOUNT=a123
+RPC_ENDPOINTS=http://127.0.0.1:8888|http://192.168.1.1:8888
+PORT=18888
+LOCK_GAS_PRICE=true
+MINER_PERMISSION="active"
+EXPIRE_SEC=60
+```
+- build and start the miner
+```
+npm install
+yarn build
+yarn start
+```
+- test if the miner is working
+```
+curl http://127.0.0.1:18888 -X POST -H "Accept: application/json" -H "Content-Type: application/json" --data '{"method":"eth_gasPrice","params":[],"id":1,"jsonrpc":"2.0"}'
+{"jsonrpc":"2.0","id":1,"result":"0x22ecb25c00"}
+```
+
+<a name="CRB"></a>
+## [For centralized exchanges] Calculate the irreversible block number from EOS (L1) chain to EOS-EVM (L2) Chain
+For centralized exchanges it is important to know up to which block number the chain is irreversible. This is the way to calculate the irreversible time of EOS-EVM:
+- ensure the leap node & eos-evm-node are fully sync-up.
+- do a get_info request to leap node.
+```
+{
+  "server_version": "943d1134",
+  "chain_id": "aca376f206b8fc25a6ed44dbdc66547c36c6c33e3a119ffbeaef943642f0e906",
+  "head_block_num": 316609050,
+...
+  "earliest_available_block_num": 302853021,
+  "last_irreversible_block_time": "2023-06-23T03:10:35.500"
+}
+```
+- in the above example all EVM blocks before `"last_irreversible_block_time": "2023-06-23T03:10:35.500"` are irreversible. Use the time conversion script:
+`
+python3 -c 'from datetime import datetime; print(hex(int((datetime.strptime("2023-06-23T03:10:35.500","%Y-%m-%dT%H:%M:%S.%f")-datetime(1970,1,1)).total_seconds())))'
+`
+to get the EVM irreversible blocktime in hex `0x64950d2b`. By scanning every EVM block, we found out that the EVM blocks up to ```6828746``` are irreversible, because its timestamp is `0x64950d2b`:
```

Comment:
Should we use the mixhash and relate to the EOS block ID to confirm the correct block was found?

Reply:
added the mixHash check. please review again.